lib/schemes/oauth2.js

```diff
@@ -171,4 +173,71 @@ export default class Oauth2Scheme {
 
     return true // True means a redirect happened
   }
+
+  // ---------------------------------------------------------------
+  // Watch axios requests for token expiration
+  // ---------------------------------------------------------------
+
+  watchTokenExpiration () {
+    const { $axios } = this.$auth.ctx.app
+    let isRefreshing = false
+
+    $axios.onRequest(async (config) => {
+      // token and refresh token
+      let token = this.$auth.getToken(this.name)
+      let refreshToken = this.$auth.getRefreshToken(this.name)
+
+      // token or "refresh token" does not exists
+      if (!token || !refreshToken || !token.length || !refreshToken.length) {
+        return config
+      }
+
+      // if already trying to refresh token, do not try again
+      if (isRefreshing) {
+        return config
+      }
+
+      // time variables
+      const tokenExpiresAt = jwtDecode(token).exp * 1000
```

Comment:
Please do not rely on access token being a JWT, but instead use the `expires_in` response returned along with `access_token`.